sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/UnboundedRabbitMqReader.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.rabbitmq;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.GetResponse;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.apache.beam.sdk.io.UnboundedSource;
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.joda.time.Instant;
+
+class UnboundedRabbitMqReader extends UnboundedSource.UnboundedReader<RabbitMqMessage> {
+
+  private final RabbitMQSource source;
+
+  private TimestampPolicy timestampPolicy;
```

Comment:
Notable changes:
- contains TimestampPolicy
- contains RecordIdPolicy
- utilizes TimestampPolicyContext for tracking "last time we checked for a new message", etc.
- no longer retain currentRecordId (calculated on the fly using RecordIdPolicy)
- no longer retain currentRecordId (calculated on the fly using TimestampPolicy)
- no longer retain watermark (calculated on the fly using TimestampPolicy)
- it is acceptable to utilize the incoming CheckpointMark. It will fail at some point when it tries to finalize a checkpoint if it had previously unacked messages. If not, everything should work fine, and we have the benefit of retaining the previous watermark this way.